modules/core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/computation/BulkScrollerComputation.java

```diff
@@ -284,17 +286,38 @@ protected void updateStatusAsScrolling(ComputationContext context, String comman
                 BulkCodecs.getStatusCodec().encode(delta));
     }
 
+    /**
+     * @deprecated since 11.5, use {@link #updateStatusAfterScroll(ComputationContext, String, String, int)} instead
+     **/
+    @Deprecated(since = "11.5", forRemoval = true)
     protected void updateStatusAfterScroll(ComputationContext context, String commandId, String errorMessage) {
-        updateStatusAfterScroll(context, commandId, 0, errorMessage, false);
+        int errorCode = StringUtils.isEmpty(errorMessage) ? 0 : HttpStatus.SC_INTERNAL_SERVER_ERROR;
+        updateStatusAfterScroll(context, commandId, errorMessage, errorCode);
+    }
+
+    protected void updateStatusAfterScroll(ComputationContext context, String commandId, String errorMessage,
+            int errorCode) {
+        updateStatusAfterScroll(context, commandId, 0, errorMessage, errorCode, false);
     }
 
     protected void updateStatusAfterScroll(ComputationContext context, String commandId, long documentCount,
             boolean limited) {
-        updateStatusAfterScroll(context, commandId, documentCount, null, limited);
+        updateStatusAfterScroll(context, commandId, documentCount, null, 0, limited);
     }
 
+    /**
+     * @deprecated since 11.5, use
+     *             {@link #updateStatusAfterScroll(ComputationContext, String, long, String, int, boolean)} instead
+     **/
+    @Deprecated(since = "11.5", forRemoval = true)
     protected void updateStatusAfterScroll(ComputationContext context, String commandId, long documentCount,
             String errorMessage, boolean limited) {
+        int errorCode = StringUtils.isEmpty(errorMessage) ? 0 : HttpStatus.SC_INTERNAL_SERVER_ERROR;
+        updateStatusAfterScroll(context, commandId, documentCount, errorMessage, errorCode, limited);
+    }
+
+    protected void updateStatusAfterScroll(ComputationContext context, String commandId, long documentCount,
+            String errorMessage, int errorCode, boolean limited) {
```

Comment:
`errorMessage` and `errorCode` should be both provided or not, we should remove cases where we can have one or the other.

As it, we could use `BulkStatus#inError(String, int)` and remove `BulkStatus#setErrorCode`.

Reply:
> errorMessage and errorCode should be both provided or not, we should remove cases where we can have one or the other.

Yep, for now I removed them. Is it ok for you. let me know wdyt